src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -414,14 +413,31 @@ public List<Course> getAllCoursesToRegister() {
     // TODO: we should rename this into courses/{courseId}/details
     @GetMapping("courses/{courseId}/for-dashboard")
     @PreAuthorize("hasRole('USER')")
-    public Course getCourseForDashboard(@PathVariable long courseId, @RequestParam(defaultValue = "false") boolean refresh) {
+    public ResponseEntity<Course> getCourseForDashboard(@PathVariable long courseId, @RequestParam(defaultValue = "false") boolean refresh) {
         long timeNanoStart = System.nanoTime();
         log.debug("REST request to get one course {} with exams, lectures, exercises, participations, submissions and results, etc.", courseId);
         User user = userRepository.getUserWithGroupsAndAuthorities();
+
         Course course = courseService.findOneWithExercisesAndLecturesAndExamsAndLearningGoalsAndTutorialGroupsForUser(courseId, user, refresh);
+        if (!authCheckService.isAtLeastStudentInCourse(course, user)) {
+            // user might be allowed to register for the course
+            try {
+                Course courseWithOrganizations = courseRepository.findSingleNotOnlineWithOrganizationsAndPrerequisitesElseThrow(courseId);
+                authCheckService.checkUserAllowedToSelfRegisterForCourseElseThrow(user, courseWithOrganizations);
+                // suppress error alert with skipAlert: true so that the client can redirect to the registration page
+                throw new AccessForbiddenAlertException(ErrorConstants.DEFAULT_TYPE, "You don't have access to this course, but you could register.", ENTITY_NAME,
+                        "noAccessButCouldRegister", true);
+            }
+            catch (AccessForbiddenException e) {
+                // user is not even allowed to self-register
+                // just normally throw the access forbidden exception
+                throw new AccessForbiddenException(ENTITY_NAME, courseId);
```

Comment:
Yes, that's intentional. I want to avoid confusion when accessing the `/for-dashboard` endpoint. I think it would be very strange if you got a 

> 403: You cannot register because XYZ

message from that endpoint because that's not what you intended to do. Instead I just keep the generic 403 error for these cases and reserve the more specific error messages about registration for the `/for-registration` and `/register` endpoints. Do you think that makes sense?

Reply:
Yes I think this makes sense 👍